managed/services/telemetry/telemetry.go

```diff
@@ -160,7 +160,47 @@ func (s *Service) DistributionMethod() serverpb.DistributionMethod {
 	return s.sDistributionMethod
 }
 
-func (s *Service) prepareReport(ctx context.Context) *reporter.ReportRequest {
+func (s *Service) processSendCh(ctx context.Context) {
+	var inflightReports []*pmmv1.ServerMetric
+	var sendCtx context.Context
+	var cancel context.CancelFunc
+
+	for {
+		select {
+		case report, ok := <-s.sendCh:
+			if ok {
+				inflightReports = append(inflightReports, report)
+				if sendCtx != nil {
+					cancel()
+				}
+				sendCtx, cancel = context.WithCancel(context.Background())
+
+				go func(ctx context.Context, reports *[]*pmmv1.ServerMetric) {
+					reportsCopy := make([]*pmmv1.ServerMetric, len(*reports))
+					copy(reportsCopy, *reports)
```

Comment:
send copy of slice, to avoid potential concurrent modification

Reply:
I think we should copy in the main goroutine avoid concurrency conflicts

Reply:
yeah, copy probably not an atomic. thanks